native/arrow_format_nif/src/message.rs

```diff
@@ -0,0 +1,18 @@
+use crate::utils::CustomMetadata;
+use rustler::{Atom, NifRecord};
+
+pub mod header;
+pub mod record_batch;
+pub mod schema;
+
+use header::Header;
+
+#[derive(Debug, NifRecord)]
+#[tag = "message"]
+pub struct Message {
+    pub version: Atom, // TODO Limit this to just v1..v5.
```

Comment:
Would [`UnitEnums`](https://docs.rs/rustler/latest/rustler/derive.NifUnitEnum.html) be good for this?

Reply:
Looks like it's exactly what you need :)

My suggestion would be basically what it accomplishes: declare an enum, and derive the encoder and decoder traits to convert to and from elixir atoms